### DISCUSSION TITLE
What's the status of Elm support in Parcel v2?

### DISCUSSION THREAD

**Discussion by tad-lispy** [parcel-bundler/parcel#5948] [Q&A] [answered]
Hi! Thanks for all your hard work on Parcel. It's wonderful.

Today I tried to switch to v2. This page <https://v2.parceljs.org/languages/elm/> seems to imply that bundling Elm source code "just works". But following the example there I'm getting:

```
$ ./node_modules/.bin/parcel --version
2.0.0-nightly.610+8a4fa3ee

$ ./node_modules/.bin/parcel src/index.html
ℹ️ Server running at http://localhost:1234
🚨 Build failed.
Error: No transformers found for "/snake-game/src/Main.elm".
Error: No transformers found for "/snake-game/src/Main.elm".
    at ParcelConfig._getTransformerNodes (/snake-game/node_modules/@parcel/core/lib/ParcelConfig.js:157:13)
    at ParcelConfig.getTransformerNames (/snake-game/node_modules/@parcel/core/lib/ParcelConfig.js:164:29)
    at ConfigLoader.loadParcelConfig (/snake-game/node_modules/@parcel/core/lib/ConfigLoader.js:82:37)
    at /snake-game/node_modules/@parcel/core/lib/ConfigLoader.js:54:48
    at processTicksAndRejections (node:internal/process/task_queues:94:5)
    at async PromiseQueue._runFn (/snake-game/node_modules/@parcel/utils/lib/PromiseQueue.js:98:7)
    at async PromiseQueue._next (/snake-game/node_modules/@parcel/utils/lib/PromiseQueue.js:85:5)
```

Am I missing something?

**Comment by tad-lispy** [comment, marked as answer]
I got it to work using nightly release of Parcel:

``` sh
rm -rf node_modules package-lock.json .parcel-cache
npm install --save-dev parcel@nightly
```

There was a little quirk. After first run I got an error, but restarting parcel  produced a desired effect.

```
root@1e1596cb588d:/snake-game# npm run develop

> snake-game@1.0.0 develop
> parcel src/index.html

ℹ️ Server running at http://localhost:1234
🚨 Build failed.
unknown: Could not resolve module "elm" from "/snake-game"
Autoinstall is disabled, please install this package manually and restart Parcel.

# Done absolutely nothing other than terminating the process with ^C

root@1e1596cb588d:/snake-game# npm run develop

> snake-game@1.0.0 develop
> parcel src/index.html

ℹ️ Server running at http://localhost:1234
✨ Built in 186ms
```

Parcel kind of says I should restart it, but should it be like that? Also, shouldn't it work with `next` release (`beta-1` as of today)? Running on nightly seems dangerous :fearful:

**Reply by mischnic** [reply]
Strange that you didn't actually have to install elm

> Running on nightly seems dangerous

No, there are more bugs in the beta than the nightly.

**Reply by tad-lispy** [reply]
I did install Elm like the documentation advised. Just didn't mention it.

I've been playing with it for few hours now and it works very well. The only confusing part was having to restart Parcel once after introducing Elm for the first time.

Again, thank you very much for developing Parcel. I was using v1 for several years now and it's always been a pleasure. I really appreciate that I don't have to maintain any configuration.
